core/block.go

```diff
@@ -37,6 +37,8 @@ type Header struct {
 	EventsBloom *bloom.BloomFilter
 	// Amount of ETH charged per Gas spent
 	GasPrice *felt.Felt
+	// Sequencer signatures
+	Signatures [][]*felt.Felt
```

Comment:
Do we only need the most recent signatures in memory for serving p2p? Why do we need to store them for every block?

Reply:
That is true, syncing peer only needs to verify the most recent signature. But storing them only in-memory is a bit too risky and contradicting to what we have done so far, which is keeping everything in database.

> do we need to store them for every block?

We need to store them for all non-L1-verified blocks. Because chain might get reorged at any point and any of those blocks could become the HEAD. 

Later on, if we end up needing to free some storage, we can remove the signatures of L1-verified blocks.